controller/src/main/java/io/pravega/controller/store/stream/records/StreamConfigurationRecord.java

```diff
@@ -301,6 +302,23 @@ private void write02(StreamConfigurationRecord streamConfigurationRecord, Revisi
             revisionDataOutput.writeCollection(streamConfigurationRecord.removeTags, stringSerializer);
         }
 
+        private void read03(RevisionDataInput revisionDataInput,
+                            StreamConfigurationRecordBuilder configurationRecordBuilder)
+                throws IOException {
+            StreamConfiguration.StreamConfigurationBuilder streamConfigurationBuilder = StreamConfiguration.builder();
+            streamConfigurationBuilder.scalingPolicy(configurationRecordBuilder.streamConfiguration.getScalingPolicy())
+                                      .retentionPolicy(configurationRecordBuilder.streamConfiguration.getRetentionPolicy())
+                                      .timestampAggregationTimeout(configurationRecordBuilder.streamConfiguration.getTimestampAggregationTimeout())
+                                      .tags(configurationRecordBuilder.streamConfiguration.getTags())
+                                      .rolloverSizeBytes(revisionDataInput.readLong());
```

Comment:
This would read corrupted data, right? 
`write02(..)` writes other fields.

Reply:
Sorry I don't quite get it. My understanding is that what's written by `write02` should be read up by the `read02` and `read03` here should only be reading what's written by `write03`. Is that the case?

Reply:
Yes, the way it's declared above, `read02` is paired with `write02` and mapped to revision 2; while `read03` is paired with `write03` and revision 3. The `VersionedSerializer` will make sure to invoke the appropriate read method for each revision.

Reply:
As far as I can see, we're now writing a single long in 03 and reading a single long in 03. This is fine. Old code will not have the 03 methods and will thus ignore this value, and the new code, when reading something serialized with the old version, will not be invoking the 03 methods at all since there's no revision 3 encoded in it (old code doesn't know about  it).

Reply:
Thanks.